### PR TITLE
net: lwm2m: Senml Json Base name dynamical update

### PR DESCRIPTION
**Pull request by jheiskan81**
- SenML Json support dynamical base name for composite operation.
- Changes simplify base name generation and compress message better.

### CODE REVIEW

**Comment by jheiskan81** [comment]
@VeijoPesonen @jarlamsa @SeppoTakalo @rlubos Please review.

File: subsys/net/lib/lwm2m/lwm2m_rw_senml_json.c [hunk 1]
```diff
@@ -291,10 +290,9 @@ static int put_begin(struct lwm2m_output_context *out, struct lwm2m_obj_path *pa
 	}
 
 	/*
-	 * Enable base name add if it is enabled
-	 * Base name is only added one time at first resource data
+	 * Init base level state for skip first object instance compare
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_rw_senml_json.c:293]: nit: it seems that this could be made a single-line comment now (`/* ... */`).

**Review by rlubos — DISMISSED** [review]
Pretty neat, just one comment.

**Review by SeppoTakalo — DISMISSED** [review]

File: subsys/net/lib/lwm2m/lwm2m_rw_senml_json.c (hunk 1 above)
Comment (jheiskan81) [reply subsys/net/lib/lwm2m/lwm2m_rw_senml_json.c:293]: I can change a comment to single line.

**Review by SeppoTakalo — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**